appengine/standard_python3/bundled-services/mail/wsgi/main.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cgi
+import http
+import os
+import re
+
+from google.appengine.api import mail
+from google.appengine.api import wrap_wsgi_app
+
+
+def HelloReceiver(environ, start_response):
+    if environ["REQUEST_METHOD"] != "POST":
+        return ("", http.HTTPStatus.METHOD_NOT_ALLOWED, [("Allow", "POST")])
+
+    message = mail.InboundEmailMessage.from_environ(environ)
+
+    print(f"Received greeting for {message.to} at {message.date} from {message.sender}")
+    for content_type, payload in message.bodies("text/plain"):
+        print(f"Text/plain body: {payload.decode()}")
+        break
+
+    response = http.HTTPStatus.OK
+    start_response(f"{response.value} {response.phrase}", [])
+    return ["success".encode("utf-8")]
+
+
+def BounceReceiver(environ, start_response):
+    if environ["REQUEST_METHOD"] != "POST":
+        return ("", http.HTTPStatus.METHOD_NOT_ALLOWED, [("Allow", "POST")])
+
+    bounce_message = mail.BounceNotification.from_environ(environ)
+
+    # Do something with the message
+    print("Received bounce post.")
+    print("Bounce original: %s", bounce_message.original)
+    print("Bounce notification: %s", bounce_message.notification)
+
+    # Return suitable response
+    response = http.HTTPStatus.OK
+    start_response(f"{response.value} {response.phrase}", [])
+    return ["success".encode("utf-8")]
+
+
+def InvalidMailSender(environ, start_response):
+    # Send invalid mail to trigger a bounce notification.
+    mail.send_mail(
+        sender="test-python-user@shreejad-knative-dev.appspotmail.com",
```

Comment:
Do we want the `shreejad-knative-dev` bit in here, or should we use a more generic "placeholder" value?

Reply:
That should not be there! I thought I'd fixed that. I've removed it and the route to it. To send a mail you want bounced, use the same method as sending a valid mail, just to a bad address. Fixed now.